aten/src/ATen/functorch/BatchRulesFactory.cpp

```diff
@@ -103,6 +103,121 @@ static std::tuple<Tensor,optional<int64_t>> _new_zeros_with_same_feature_meta_ba
   return std::make_tuple(result, 0);
 }
 
+static std::tuple<Tensor,optional<int64_t>> linspace_logspace_batch_rule_helper(
+    const at::Tensor& start, optional<int64_t> start_bdim,
+    const at::Tensor& end, optional<int64_t> end_bdim,
+    int64_t steps,
+    c10::optional<double> base,
+    c10::optional<at::ScalarType> dtype,
+    c10::optional<at::Layout> layout,
+    c10::optional<at::Device> device,
+    c10::optional<bool> pin_memory)
+{
+  auto batch_size = get_bdim_size2(start, start_bdim, end, end_bdim);
+  auto start_ = ensure_has_bdim(start, start_bdim.has_value(), batch_size);
+  auto end_ = ensure_has_bdim(end, end_bdim.has_value(), batch_size);
+  start_ = moveBatchDimToFront(start_, start_bdim);
+  end_ = moveBatchDimToFront(end_, end_bdim);
+
+  auto tensor_options = at::TensorOptions().dtype(dtype).layout(layout).device(device).pinned_memory(pin_memory);
+
+  Tensor result;
+  if (steps == 0){
+    result = at::full({batch_size, 0}, 0, tensor_options);
+  } else if (steps == 1){
+    result = at::empty({batch_size}, tensor_options).copy_(start_).unsqueeze(1);
```

Comment:
could we do `start_.new_empty({batch_size}, tensor_options)`? If start_ is being vmapped over I am not sure the current code would work

Reply:
Yes, the tests passed.